var/spack/repos/builtin/packages/ucc/package.py

```diff
@@ -55,5 +60,20 @@ def configure_args(self):
         args = []
         args.extend(self.with_or_without("cuda", activation_value="prefix"))
         args.extend(self.with_or_without("nccl", activation_value="prefix"))
-        # args.extend(self.with_or_without("rccl", activation_value="prefix"))
+        if self.spec.satisfies("+rocm"):
+            rocm_flags = " ".join(
+                [
+                    "-I" + self.spec["hip"].prefix.include,
+                    "-I" + self.spec["hip"].prefix.include.hip,
+                    "-I" + self.spec["hsa-rocr-dev"].prefix.include.hsa,
+                    "-L" + self.spec["hip"].prefix.lib,
+                    "-L" + self.spec["hsa-rocr-dev"].prefix.lib,
```

Comment:
These should be already exposed through dependencies.

Reply:
If these are not set then ucc will assume that hip and hsa-rocr-dev are in a single directory like **/opt/rocm** and set the -I/-L flags accordingly. That is not the case with spack-built ROCm as both hsa and hip are in different paths. This is why we have to use ROCM_PARSE_FLAGS function in the rocm.m4 file. However, it seems the function hasn't been implemented properly which is why I had to create the patch.

Reply:
I've added these flags to the cppflags and ldflags instead of using the ROCM_PARSE_FLAGS function.